wagtail/snippets/action_menu.py

```diff
@@ -77,6 +78,121 @@ def is_shown(self, context):
         )
 
 
+class SubmitForModerationMenuItem(ActionMenuItem):
+    name = "action-submit"
+    label = _("Submit for moderation")
+    icon_name = "resubmit"
+
+    def is_shown(self, context):
```

Comment:
Yes, I think these share a huge amount of logic - it would be really good to reduce that duplication. I do wonder if you could could just refactor out the few page-specific checks into separate methods quite easily here?

Reply:
The classes are currently not inherited nor have any shared templates whatsoever with the ones from pages, so even if refactored into separate methods, it still requires a fair amount of other refactoring to share the logic.